payload-functions.js

```diff
@@ -183,24 +183,18 @@ function prepareDeleteSubnetPayload(params) {
 
 function validateAddSecurityGroupRulesParams(params) {
   if (params.ipProtocol !== "ICMP" && params.ipProtocol !== "All") {
-    if (!params.fromPorts?.length || !params.toPorts?.length) {
-      throw new Error(`From Ports and To Ports are required for ${params.ipProtocol} protocol.`);
+    if (!params.portRanges?.length) {
```

Comment:
That will trigger if the field is empty, which we can validate with `config.json`, or am I missing something?

Reply:
This field can't be empty only when the selected protocol is "TCP" or "UDP", so we can't add `"required": true` in **config.json**.